src/main/java/org/mule/service/oauth/internal/AbstractOAuthDancer.java

```diff
@@ -183,8 +184,12 @@ protected TokenResponse invokeTokenUrl(String tokenUrl, Map<String, String> toke
                       new TypedValue(body, DataType.builder().fromObject(body)
                           .mediaType(responseContentType).build()))
 
-          .addBinding("attributes", new TypedValue(Collections.singletonMap("headers", headers.toImmutableParameterMap()),
-                                                   DataType.fromType(Map.class)))
+          .addBinding("attributes", new TypedValue(Collections.singletonMap("headers", headers.toImmutableMultiMap()),
+                                                   DataType.builder()
```

Comment:
Just realised this is wrong, you were setting the data type of the attributes not the headers. My bad.